### PR TITLE
feat: Parametrize createSubscribe on client type

### PR DESCRIPTION
**Pull request by taion**

### CODE REVIEW

**Comment by taion** [comment]
need this to extend the client in a type-safe way

File: src/index.ts [hunk 1]
```diff
@@ -1,17 +1,32 @@
...
+// FIXME: These should be type imports.
+import { TokenResponse, TokenStorage } from './LocalTokenStorage';
+import { Network } from './NetworkLayer';
+import { FetchOptions } from './createFetch';
+import {
+  SubscriptionClientOptions,
+  SubscriptionOptions,
+} from './createSubscribe';
+import { UseAuthTokenOptions } from './useAuthToken';
+/* eslint-enable import/no-duplicates, import/order */
```
Comment (taion) [inline src/index.ts]: The previous way didn't work with generics.

TS 3.8 may support `import type` and `export type`, but neither `@typescript-eslint/eslint-plugin` nor Prettier do!

File: src/index.ts [hunk 2]
```diff
@@ -1,17 +1,32 @@
...
-export type TokenStorage = import('./LocalTokenStorage').TokenStorage;
+// FIXME: These should be type imports.
+import { TokenResponse, TokenStorage } from './LocalTokenStorage';
+import { Network } from './NetworkLayer';
+import { FetchOptions } from './createFetch';
+import {
+  SubscriptionClientOptions,
+  SubscriptionOptions,
+} from './createSubscribe';
+import { UseAuthTokenOptions } from './useAuthToken';
```
Comment (taion) [inline src/index.ts]: Though... why do we export these types anyway? Are they useful?

File: src/createSubscribe.ts [hunk 3]
```diff
@@ -152,3 +161,22 @@ export default function createSubscribe({
...
   return subscribeFn;
 }
+
+function createSubscribe(
+  options?: SubscriptionClientOptions,
+): SubscribeFunction<SocketIoSubscriptionClient>;
+function createSubscribe<TSubscriptionClient extends SubscriptionClient>(
+  options: SubscriptionOptions<TSubscriptionClient>,
+): SubscribeFunction<TSubscriptionClient>;
+function createSubscribe(options: any): any {
```
Comment (taion) [inline src/createSubscribe.ts]: Is this right?

**Comment by taion** [comment]
@jquense ping – thoughts?

File: src/createSubscribe.ts (hunk 3 above)
Comment (jquense) [reply src/createSubscribe.ts]: for overloads? looks right. Does it work? I think i tried something like this and it didn't immediately work so i bailed.

**Pull request merged**